include/irr/asset/CTexturePacker.h

```diff
@@ -60,21 +121,13 @@ class CTexturePacker
         auto extent = _img->getCreationParameters().extent;
         uint32_t rnd = core::roundUpToPoT(std::max(extent.width,extent.height))/m_pgSzxy;
         uint32_t pgCnt = rnd*rnd;
-        uint32_t morton_addr = m_addrAlctr.alloc_addr(pgCnt, pgCnt);
-        assert(morton_addr!=addr_alctr_t::invalid_address);
-
-        auto dec_morton = [](uint32_t x) -> uint32_t {
-            x = x & 0x55555555u;
-            x = (x | (x >> 1)) & 0x33333333u;
-            x = (x | (x >> 2)) & 0x0F0F0F0Fu;
-            x = (x | (x >> 4)) & 0x00FF00FFu;
-            x = (x | (x >> 8)) & 0x0000FFFFu;
-            return x;
-        };
-        core::vector2du32_SIMD texOffset(dec_morton(morton_addr), dec_morton(morton_addr >> 1));
-        texOffset *= m_pgSzxy;
+        uint32_t addr = m_addrAlctr.alloc_addr(pgCnt, pgCnt);
+        uint32_t addr_layer = m_addrAlctr.unpackAddress_layer(addr);
+        assert(addr!=addr_alctr_t::invalid_address);
+
+        const core::vector2du32_SIMD texOffset = pageCoords(m_addrAlctr.unpackAddress_x(addr), m_addrAlctr.unpackAddress_y(addr), m_pgSzxy);
         const uint32_t texelSz = getTexelOrBlockBytesize(m_megaimg->getCreationParameters().format);
-        const uint64_t bufOffset = static_cast<uint64_t>(texOffset.y * m_megaimg->getCreationParameters().extent.width + texOffset.x) * texelSz;
+        const uint64_t bufOffset = static_cast<uint64_t>(m_megaimg->getCreationParameters().extent.width*m_megaimg->getCreationParameters().extent.height*addr_layer + texOffset.y*m_megaimg->getCreationParameters().extent.width + texOffset.x) * texelSz;
 
         for (const auto& reg : _img->getRegions())
         {
```

Comment:
I think @AnastaZIuk has made a utility function for copying between layers specified as regions of two textures and uses it in GLI Loader/Writer.

Reply:
individual tiles should be copied one-by-one into the PhysicalAddressTexture